Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -1358,6 +1362,30 @@ void MainWindow::SetStateSlot(int slot)
                        2500);
 }
 
+void MainWindow::IncrementSelectStateSlot()
+{
+  int state_slot = Settings::Instance().GetStateSlot();
+  state_slot++;
+  if (state_slot > 10)
+  {
+    state_slot = 1;
+  }
+  m_state_slot = state_slot;
+  m_menu_bar->SetStateSlot(m_state_slot);
```

Comment:
I think you're overcomplicating this, it should be enough to just do

```
  int state_slot = m_state_slot + 1;
  if (state_slot > 10)
    state_slot = 1;
  SetStateSlot(state_slot);
```

(similar for the decrement case)

Reply:
You probably should also use `State::NUM_STATES` instead of `10` for the max state number.

Reply:
Sounds good. I made the changes to the code per your recommendations.